ic-agent/src/identity/mod.rs

```diff
@@ -23,8 +23,6 @@ pub trait Identity: Send + Sync {
     fn sender(&self) -> Result<Principal, String>;
 
     /// Sign a blob, the concatenation of the domain separator & request ID,
-    /// creating the sender signature, with the principal passed in.
-    /// The principal should be
-    /// the same returned by the call to `sender()`.
-    fn sign(&self, blob: &[u8], principal: &Principal) -> Result<Signature, String>;
+    /// creating the sender signature.
+    fn sign(&self, blob: &[u8]) -> Result<Signature, String>;
```

Comment:
This is a `BREAKING CHANGE` and should have a `# BREAKING CHANGE` section in the description.

Reply:
Also see my comment above. I do have a use case :)